### PR TITLE
Instance Scheduler resources

### PR DESCRIPTION
**Pull request by gfou-al**
Add IAM role and lambda function deployment for the Instance Scheduler Lambda Function. Use the latest tag.

### CODE REVIEW

**Review by davidkelliott — DISMISSED** [review]
Looks good (noting to change role to more restrictive one after testing deployment)

File: terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+resource "aws_lambda_function" "instance-scheduler-lambda-function" {
+  #checkov:skip=CKV_AWS_116
+  #checkov:skip=CKV_AWS_117
+  #checkov:skip=CKV_AWS_272 "Code signing not required"
+  function_name                  = "instance-scheduler-lambda-function"
+  handler                        = "main"
+  reserved_concurrent_executions = 0
+  runtime                        = "go1.x"
```
Comment (ewastempel) [inline terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf:8]: can we lock the version more? what do we use in other scripts?

File: terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  image_uri                      = "${module.instance_scheduler_ecr_repo.ecr_repository_name}:latest"
```
Comment (ewastempel) [inline terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf:9]: latest tag should be default, but ideally we want to pass this as a variable, don't we?

**Review by davidkelliott — DISMISSED** [review]

**Review by ewastempel — CHANGES_REQUESTED** [review]
the static code analysis are failing now for: https://github.com/ministryofjustice/modernisation-platform/actions/runs/3243700870/jobs/5318796513

File: terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf (hunk 1 above)
Comment (gfou-al) [reply terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf:8]: I don't think we can. According to https://docs.aws.amazon.com/lambda/latest/dg/API_CreateFunction.html#SSS-CreateFunction-request-Runtime at the moment `go1.x` seems to be the only valid value for the runtime configuration for the Go language.

File: terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf (hunk 2 above)
Comment (gfou-al) [reply terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf:9]: We use terraform variables to pass different values to terraform modules. In this case, the lambda is not part of a module and only deployed as a single instance in core-shared-services. I don't see why we would be using a variable here, but let me know if you could point me to an example in the existing code.

**Comment by gfou-al** [comment]
> the static code analysis are failing now for: https://github.com/ministryofjustice/modernisation-platform/actions/runs/3243700870/jobs/5318796513

Done

File: terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf (hunk 2 above)
Comment (ewastempel) [reply terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf:9]: cool ,thanks for the explanation

Comment (ewastempel) [reply terraform/environments/core-shared-services/instance-scheduler-lambda-function.tf:9]: ok, let's tackle it in the future

**Review by ewastempel — APPROVED** [review]
Cool

**Pull request merged**